src/components/svg-canvas-graph.tsx

```diff
@@ -92,24 +99,91 @@ const SvgCanvas = () => {
                 dispatch(addSelected(node));
             }
         }
+        dispatch(setPolyLines(getPolylines(graph.current)));
         // console.log('down ', graph.current.getNodeAttributes(node));
     });
     const handlePointerMove = useEvent((node: StnId | MiscNodeId, e: React.PointerEvent<SVGElement>) => {
         const { x, y } = getMousePosition(e);
 
         if (mode === 'free' && active === node) {
-            selected.forEach(s => {
-                if (graph.current.hasNode(s)) {
-                    graph.current.updateNodeAttributes(s, attr => ({
-                        ...attr,
-                        x: roundToNearestN(attr.x - ((offset.x - x) * svgViewBoxZoom) / 100, e.altKey ? 1 : 5),
-                        y: roundToNearestN(attr.y - ((offset.y - y) * svgViewBoxZoom) / 100, e.altKey ? 1 : 5),
-                    }));
+            if (!e.altKey) {
+                // use polyline if alt key is not pressed
+                const fromX = graph.current.getNodeAttribute(node, 'x');
+                const fromY = graph.current.getNodeAttribute(node, 'y');
+                const toX = fromX - ((offset.x - x) * svgViewBoxZoom) / 100;
+                const toY = fromY - ((offset.y - y) * svgViewBoxZoom) / 100;
+
+                let newX = toX;
+                let newY = toY;
+
+                if (isNodeSupportPolyline(node, graph.current)) {
+                    if (activePolylines.length !== 0) {
+                        // remove the active polylines if the distance is larger than 5
+                        setActivePolylines(activePolylines.filter(l => getPolylineDistance(l, toX, toY) <= 5));
+                    }
+                    if (activePolylines.length < 2) {
+                        // add the nearest polylines if the distance is less than 3
```

Comment:
This comment tells the same thing the next line of code tells. Comments should be the reason why we do this not what we are doing.

https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/